### PR TITLE
[inductor] correctly retrieve the "shared" attribute from a Triton binary

### PR DESCRIPTION
**Pull request by chenyang78**
Stack from [ghstack](https://github.com/ezyang/ghstack) (oldest at bottom):
* __->__ #120666

cc @voznesenskym @penguinwu @EikanWang @jgong5 @Guobing-Chen @XiaobingSuper @zhuhaozhe @blzheng @wenzhe-nrv @jiayisunx @peterbell10 @ipiszy @yf225 @kadeng @muchulee8 @aakhundov @ColinPeppler @amjames

### CODE REVIEW

File: torch/_inductor/triton_heuristics.py [hunk 1]
```diff
@@ -422,7 +423,7 @@ def launcher({', '.join(def_args)}, grid, stream):
         launcher.config = cfg
         launcher.n_regs = getattr(binary, "n_regs", None)
         launcher.n_spills = getattr(binary, "n_spills", None)
-        launcher.shared = getattr(binary, "shared", None)
+        launcher.shared = binary_shared
```
Comment (aakhundov) [inline torch/_inductor/triton_heuristics.py:426]: Wondering why not just `s/binary/scope/` here?

File: torch/_inductor/triton_heuristics.py [hunk 2]
```diff
@@ -422,7 +423,7 @@ def launcher({', '.join(def_args)}, grid, stream):
         launcher.config = cfg
         launcher.n_regs = getattr(binary, "n_regs", None)
         launcher.n_spills = getattr(binary, "n_spills", None)
-        launcher.shared = getattr(binary, "shared", None)
```
Comment (aakhundov) [inline torch/_inductor/triton_heuristics.py:425]: Wondering why not just `s/binary/scope/` here?

**Review by jansel — APPROVED** [review]

File: torch/_inductor/triton_heuristics.py (hunk 2 above)
Comment (chenyang78) [reply torch/_inductor/triton_heuristics.py:425]: I think we doesn't really need to run an extra search via ```getattr```. Seems assigning a previously-retrieved value is slightly more straightforward :)

**Pull request closed without merging**